activerecord/lib/active_record/core.rb

```diff
@@ -487,12 +493,17 @@ def connection_handler
 
     # Returns the contents of the record as a nicely formatted string.
     def inspect
+      filter_attributes = self.filter_attributes.map(&:to_s)
```

Comment:
Since this will have `include?` called on it for each attribute in the model, I think we should convert it to a set (`include?` is linear time for an array, but constant time for a set).

It might also be worth doing the conversion on assignment, as we do for `ignored_columns`: https://github.com/rails/rails/blob/e6ba30efbf107954db2707410d78d77680d69995/activerecord/lib/active_record/model_schema.rb#L274-L276